source/_integrations/mqtt.markdown

```diff
@@ -301,14 +301,242 @@ The discovery topic needs to follow a specific format:
 - `<object_id>`: The ID of the device. This is only to allow for separate topics for each device and is not used for the `entity_id`. The ID of the device must only consist of characters from the character class `[a-zA-Z0-9_-]` (alphanumerics, underscore and hyphen).
 
 The `<node_id>` level can be used by clients to only subscribe to their own (command) topics by using one wildcard topic like `<discovery_prefix>/+/<node_id>/+/set`.
 
 Best practice for entities with a `unique_id` is to set `<object_id>` to `unique_id` and omit the `<node_id>`.
 
+#### Device discovery payload
+
+A device can send a discovery payload to expose all components for a device.
+The `<component>` part in the discovery topic must be set to `device`.
+
+As an alternative it is also possible a device [can send a discovery payload for each component](/integrations/mqtt/#single-component-discovery-payload) it wants to set up.
+
+The shared options are at root level of the JSON message and must include the `device` mapping (abbreviated as `dev`) and `origin` mapping (abbreviated as `o`). The `device` and `origin` mappings are required options and cannot be overridden at entity/component level.
+
+Supported shared options are:
+
+- The `availability` [options](/integrations/mqtt/#using-availability-topics).
+- The `origin` (required) [options](/integrations/mqtt/#adding-information-about-the-origin-af-a-discovery-message)
+- `command_topic`
+- `state_topic`
+- `qos`
+- `encoding`
+
+The component specific options are placed as mappings under the `components` key (abbreviated as `cmp`) like:
+
+```json
+{
+  "dev": {
+    "ids": "ea334450945afc",
+    "name": "Kitchen",
+    "mf": "Bla electronics",
+    "mdl": "xya",
+    "sw": "1.0",
+    "sn": "ea334450945afc",
+    "hw": "1.0rev2",
+  },
+  "o": {
+    "name":"bla2mqtt",
+    "sw": "2.1",
+    "url": "https://bla2mqtt.example.com/support",
+  },
+  "cmp": {
+    "some_unique_component_id1": {
+      "platform": "sensor",
+      "device_class":"temperature",
+      "unit_of_measurement":"°C",
+      "value_template":"{% raw %}{{ value_json.temperature}}{% endraw %}",
+      "unique_id":"temp01ae_t",
+    },
+    "some_unique_id2": {
+      "platform": "sensor",
+      "device_class":"humidity",
+      "unit_of_measurement":"%",
+      "value_template":"{% raw %}{{ value_json.humidity}}{% endraw %}",
+      "unique_id":"temp01ae_h",
+    }
+  },
+  "state_topic":"sensorBedroom/state",
+  "qos": 2,
+}
+```
+
+The components id's under the `components` (`cmp`) key should are used as part of the discovery identification. A `platform` config option is required for each component config that is added to identify the component platform.
+
+To remove the components publish an empty string to the discovery topic. This will remove the component and clear the published discovery payload. It will also remove the device entry if there are no further references to it.
```

Comment:
```suggestion
To remove the components, publish an empty string to the discovery topic. This will remove the component and clear the published discovery payload. It will also remove the device entry if there are no further references to it.
```